simplesat/pool.py

```diff
@@ -29,15 +34,15 @@ def add_repository(self, repository):
         repository : Repository
             The repository to add
         """
+        self._repositories.append(repository)
         for package in repository:
             current_id = self._id
             self._id += 1
+            self._id_to_package_[current_id] = package
+            self._package_to_id_[package] = current_id
+            self._packages_by_name_[package.name].append(package)
 
-            self._id_to_package[current_id] = package
-            self._package_to_id[package] = current_id
-            self._packages_by_name[package.name].append(package)
-
-    def what_provides(self, requirement):
+    def what_provides(self, requirement, transform=True):
```

Comment:
can we rename `transform` into e.g. `use_modifiers` ? This way it is clearer what transform refers to IMO